### DISCUSSION TITLE
No such module error when using an external library with Cocoapod

### DISCUSSION THREAD

**Discussion by amsdamsgram** [callstack/react-native-builder-bob#430] [Q&A] [answered]
Hi,

I'm trying to use a pod in my RN library but I'm getting "No such module error" in XCode.

This is what I did:
1. `npx create-react-native-library@latest awesome-library`
2. Run `pod init` in "awesome-library/ios"
4. Add `pod "HelloLib"` in `Podfile` (I'm using this pod to test)
5. Run `pod install`
6. Add `import HelloLib` in "AwesomeLibrary.swift" at the top of the file
7. Run `yarn install` then `pod install` in the example project
8. Run `yarn ios` to build the example project on the simulator

I also tried to `s.dependency = "HelloLib"` in the `.podspec` file but same error.

Am I missing something?

Thank you for your help!

**Comment by atlj** [comment]
Hey you don't have to init the pod by yourself. It's already included with native libraries. If you are trying to add a third party dependency to your library, you can add it to your `.podspec` file instead of the `Podfile`.

You should be able to add a dependency to your `.podspec` like this:

```
s.dependency = "ThirdPartyPod"
```

**Reply by amsdamsgram** [reply]
Hi @atlj, thank you for your response!

Yes, I'm trying to add a third party dependency to my library.

I've tried to add Lottie for example so I've:
1. Added `s.dependency = "lottie-ios"` in the `.podspec` of the library
2. Clean the library
3. Added `import Lottie` in the Swift file of the example's project (I'm using the `.xcworkspace` file)
4. Clean the example project
5. Build the example project
6. Then I get the error `No such module 'Lottie'`

Thank you for your help!

**Reply by amsdamsgram** [reply, marked as answer]
I finally made it work!
I've imported a `xcframework` file instead of using a pod and added `s.vendored_frameworks = 'ios/myFramework.xcframework'` in `.podspec` file as you mentionned.

Thanks for your help!
